prov/util/src/util_cq.c

```diff
@@ -61,13 +61,31 @@ int ofi_cq_write_error(struct util_cq *cq,
 
 int ofi_cq_write_error_peek(struct util_cq *cq, uint64_t tag, void *context)
 {
-	struct fi_cq_err_entry err_entry = {0};
+	struct fi_cq_err_entry err_entry = {
+		.op_context	= context,
+		.flags		= FI_TAGGED | FI_RECV,
+		.tag		= tag,
+		.err		= FI_ENOMSG,
+		.prov_errno	= -FI_ENOMSG,
+	};
+	return ofi_cq_write_error(cq, &err_entry);
+}
 
-	err_entry.op_context    = context;
-	err_entry.flags         = FI_TAGGED | FI_RECV;
-	err_entry.tag		= tag;
-	err_entry.err           = FI_ENOMSG;
-	err_entry.prov_errno    = -FI_ENOMSG;
+int ofi_cq_write_error_trunc(struct util_cq *cq, void *context, uint64_t flags,
+			     size_t len, void *buf, uint64_t data, uint64_t tag,
+			     size_t olen)
+{
+	struct fi_cq_err_entry err_entry = {
+		.op_context	= context,
+		.flags		= flags,
+		.len		= len,
+		.buf		= buf,
+		.data		= data,
+		.tag		= tag,
+		.olen		= olen,
+		.err		= FI_ETRUNC,
+		.prov_errno	= -FI_ETRUNC,
+	};
```

Comment:
We could just use ofi_cq_write_error since it's setting almost all fields.

Reply:
I'm guessing these extra calls just get inlined anyway.

Reply:
Yes, we could just use `ofi_cq_write_error()`, but from my point of view such approach (the same as for peek routine) improves readability. If we will use this function in other providers (e.x. RxD).
if any part of a provider code can be a part of utility code, I vote to keep it there.

I like both approaches.
If you would like to not change utility code -- No problem, I can do it. Thanks!